include/dlaf/tile.h

```diff
@@ -116,11 +132,23 @@ class Tile {
     return ld_;
   }
 
+  /// @brief Sets the promise to which this Tile will be moved on destruction.
+  /// @pre The promise should not be already set.
+  template <class U = T>
+  std::enable_if_t<!std::is_const<U>::value && std::is_same<T, U>::value, Tile&> setPromise(
```

Comment:
@albestro commented:
> Do we really want to expose this in the public interface?
> 
> I think that `setPromise` should be called with caution, so I would go for managing it internally without allowing anyone who has a `Tile` to be able to call it: if it is called multiple times on the same `Tile` it is a problem.

Two options are available here:
1. make it private and make the matrix class `friend`
2. make it public

Besides some small issues Option 1 has a huge problem.
It expose all the private members (which do not ensure constness of the data in case of `Tile<const T>`) to the friend classes.

To make it more safe the assertion can be replaced by a check and throw, to have the check also in Release mode.